packages/marko-web-native-x/browser/track-end-of-content.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <div id="native-x-end-of-content" />
+</template>
+
+<script>
+import { get } from '@parameter1/base-cms-object-path';
+import { endOfContent } from '../utils/gtm-events';
+
+export default {
+  props: {
+    story: {
+      type: Object,
+      required: true,
+    },
+    path: {
+      type: String,
+      required: true,
+    },
+    preview: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  created() {
+    this.observer = new IntersectionObserver((event) => {
+      if (event[0].isIntersecting) {
+        endOfContent(window);
+      }
+    });
+    const dl = window.dataLayerNativeX || [];
```

Comment:
This should detect the presence of the data layer. If it doesn't exist, nothing should happen... a new queue shouldn't be created